feat: Allow read from stream

m7kss1 opened the pull request.

m7kss1 committed:
feat: allow read from stream in cxx api

joseph-isaacs commented:
Mind signing the commits to make (DCO https://github.com/vortex-data/vortex/pull/4906/checks?check_run_id=52383374445) happy

joseph-isaacs commented:
We plan on in the future using `vortex-ipc` to stream serialized vortex arrays, however for now we don't have so I think adding a way to create a VortexFile using a byte array seems reasonable.

m7kss1 committed:
DCO Remediation Commit for Dergousov Maxim <dergousovmaxim99@gmail.com>

I, Dergousov Maxim <dergousovmaxim99@gmail.com>, hereby add my Signed-off-by to this commit: 928c8b1c7ffc2a6f0f6271531b721c22ed2b9038

Signed-off-by: Dergousov Maxim <dergousovmaxim99@gmail.com>

0ax1 commented on vortex-cxx/cpp/src/file.cpp:
```diff
@@ -8,6 +8,15 @@
 
 namespace vortex {
 
+VortexFile VortexFile::Open(const uint8_t *data, size_t length) {
```
`std::span` would be preferable here, though we currently compile with C++17 (span was introduced with C++20). @joseph-isaacs We should discuss bumping to C++20 or considering a library like abseil in order to use span. This is of course not in scope of this PR.

m7kss1 committed:
fix: lint check

Signed-off-by: Dergousov Maxim <dergousovmaxim99@gmail.com>

m7kss1 committed:
Merge branch 'develop' into cxx/allow_open_stream

m7kss1 committed:
Merge branch 'develop' into cxx/allow_open_stream

joseph-isaacs reviewed (approved):
Thanks for contribution
